### PR TITLE
Migrate checkfiles tests

### PR DESCRIPTION
**Pull request by sebasfalcone**
|Related issue|
|---|
| #26528 and #26529 |

## Description
This PR migrates the check_files tests from Jenkins into Github Actions

## Testing

- https://github.com/wazuh/wazuh/pull/26754#issuecomment-2476150759

## Extras
- 90% of the files analyzed are located in `/var/ossec/framework/python`
  - The @wazuh/devel-pyserver team should validate if it's needed to validate them (we will reduce the number of entries to 811 if removed)

### CODE REVIEW

**Comment by sebasfalcone** [comment]
## Validation matrix
| System | Architecture | Working |
|--------|--------------|---------|  
| rpm    | amd64        |   🟢    |
| rpm    | aarch64      |   🔴    |
| deb    | amd64        |   🟢    |

**Comment by sebasfalcone** [comment]
## Validation matrix

| System | Architecture | Working |
|--------|--------------|---------|
| rpm    | amd64        |   🟢    |
| rpm    | x86_64       |   🟢    |
| rpm    | aarch64      |   🟢    |
| rpm    | arm64        |   🟢    |
| deb    | x86_64       |   🟢    |
| deb    | amd64        |   🟢    |
| deb    | aarch64      |   🔴    |
| deb    | arm64        |   🔴    |

> [!IMPORTANT]
> Failing check is due to the aarch64 and arm64 using a different OS (debian 7 vs debian 9). Is this expected?

## Post migration issues
It was discovered that there is a slight variation between RPM and DEB **client-keys** group:
- **DEB**: `/var/ossec/etc/client.keys,wazuh,wazuh,640,file,-rw-r-----,,`
- **RPM**: `/var/ossec/etc/client.keys,root,wazuh,640,file,-rw-r-----,,`

Pending issue to address the problem

File: .github/workflows/packages-build-manager.yml [hunk 1]
```diff
@@ -90,8 +96,20 @@ on:
...
         required: false
 
+  # TODO: Adapt inputs for pull_request
+  # pull_request:
+  #   types: [opened, reopened, synchronize, ready_for_review]
+  #   paths:
+  #     - 'src/**'
+  #     - 'packages/**'
+  #     - '.github/workflows/packages-build-manager.yml'
+
```
Comment (LucioDonda) [inline .github/workflows/packages-build-manager.yml:106]: Do we need this ?

File: .github/actions/check_files/action.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    uses: actions/upload-artifact@v4
+    with:
+      name: validation_results
+      path: ${{ inputs.report_file }}
+
+  - name: Generate report
+    if: always()
+    shell: bash
+    run: |
+      cat ${{ inputs.report_file }} >> $GITHUB_STEP_SUMMARY
```
Comment (LucioDonda) [inline .github/actions/check_files/action.yml:96]: both steps can be attached to the compare results, right?

**Review by LucioDonda — APPROVED** [review]
LGTM, only two non-blocking comments.

File: .github/actions/check_files/action.yml (hunk 2 above)
Comment (sebasfalcone) [reply .github/actions/check_files/action.yml:96]: Yes you are right

File: .github/workflows/packages-build-manager.yml (hunk 1 above)
Comment (sebasfalcone) [reply .github/workflows/packages-build-manager.yml:106]: It will be addressed in another issue

**Pull request merged**